shotover-proxy/tests/cassandra_int_tests/collections.rs

```diff
@@ -350,42 +546,130 @@ mod map {
     }
 
     async fn insert(connection: &CassandraConnection) {
-        insert_table(connection, "map_native", |s| format!("{{ 0: {s} }}")).await;
-        insert_table(connection, "map_set", |s| format!("{{ 0: {{{s}}} }}")).await;
-        insert_table(connection, "map_list", |s| format!("{{ 0: [{s}] }}")).await;
-        insert_table(connection, "map_map", |s| format!("{{ 0: {{ 0: {s} }} }}")).await;
+        insert_table(connection, "map_native", |elements| {
+            let mut map = String::from("{");
+
+            for (i, element) in elements.iter().enumerate().take(2) {
+                map.push_str(&format!("{}: {}, ", i, element));
+            }
+            map.push_str(&format!("{}: {}}}", 2, elements[2]));
+
+            map
+        })
+        .await;
+        insert_table(connection, "map_set", |elements| {
+            let sets: String = format!("{{{}}}", elements.join(","));
+            let mut map = String::from("{");
+
+            for i in 0..2 {
+                map.push_str(&format!("{}: {}, ", i, sets));
+            }
+            map.push_str(&format!("{}: {}}}", 2, sets));
+
+            map
+        })
+        .await;
+        insert_table(connection, "map_list", |elements| {
+            let lists = format!("[{}]", elements.join(","));
+            let mut map = String::from("{");
+
+            for i in 0..2 {
+                map.push_str(&format!("{}: {}, ", i, lists));
+            }
+            map.push_str(&format!("{}: {}}}", 2, lists));
```

Comment:
would this do the same thing?
```rust
            for i in 0..3 {
                map.push_str(&format!("{}: {}, ", i, lists));
            }
            map.push_str("}");
```

Reply:
That would leave a trailing comma, assumed invalid syntax but I can double check.

Reply:
oh its that trailing comma, isnt it?

Reply:
How does it work for maps?

Reply:
I'll give it a go and report back.

Reply:
Here it is for map_list:

```rust
        insert_table(connection, "map_list", |elements| {
            let lists = format!("[{}]", elements.join(","));
            format!("{{{}}}", (0..3).map(|i| format!("{i}: {lists}")).join(","))
        })
        .await;
```

I also edited the example for set_set that I gave, there is hardcoded number of 3 outer sets that has no relation to the len of elements.